library/src/blas2/rocblas_trsv.hpp

```diff
@@ -0,0 +1,924 @@
+                                         stride_B,
+                                         &one<T>,
+                                         x_temp,
+                                         0,
+                                         1,
+                                         stride_X,
+                                         batch_count);
+            }
+
+            rocblas_gemv_template<T>(handle,
+                                     transA,
+                                     BLOCK,
+                                     BLOCK,
+                                     &one<T>,
+                                     invA,
+                                     j * BLOCK * BLOCK,
+                                     BLOCK,
+                                     stride_invA,
+                                     (U)x_temp,
+                                     0,
+                                     1,
+                                     stride_X,
+                                     &zero<T>,
+                                     B,
+                                     j * BLOCK * incx,
+                                     incx,
+                                     stride_B,
+                                     batch_count);
+        }
+
+        return rocblas_status_success;
+    }
+
+    template <rocblas_int BLOCK, bool BATCHED, typename T, typename U>
+    rocblas_status rocblas_trsv_template_mem(rocblas_handle handle,
+                                             rocblas_int    m,
+                                             rocblas_int    batch_count,
+                                             void**         mem_x_temp,
+                                             void**         mem_x_temp_arr,
+                                             void**         mem_invA,
+                                             void**         mem_invA_arr,
+                                             U              supplied_invA      = nullptr,
+                                             rocblas_int    supplied_invA_size = 0)
+    {
+
+        // Whether size is an exact multiple of blocksize
+        const bool exact_blocks = (m % BLOCK) == 0;
+
+        // perf_status indicates whether optimal performance is obtainable with available memory
+        rocblas_status perf_status = rocblas_status_success;
+
+        size_t invA_bytes   = 0;
+        size_t c_temp_bytes = 0;
+
+        // For user-supplied invA, check to make sure size is large enough
+        // If not large enough, indicate degraded performance and ignore supplied invA
+        if(supplied_invA && supplied_invA_size / BLOCK < m)
+        {
+            static int msg = fputs("WARNING: TRSV invA_size argument is too small; invA argument "
+                                   "is being ignored; TRSV performance is degraded\n",
+                                   stderr);
+            perf_status    = rocblas_status_perf_degraded;
+            supplied_invA  = nullptr;
```

Comment:
I might have missed it, but this check should be in either the _impl functions or the _template as well to set supplied_invA = nullptr for the actual execution aside from the memory management.